### PR TITLE
Improve performance of Java tests

### PR DESCRIPTION
**Pull request by samuel-rufi**

### CODE REVIEW

File: client/bindings/java/iota-client-java/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+## 1.0.0-rc.2 - 2022-10-17
+
+### Added
+- Allow integers values in QueryParams;
```
Comment (thibault-martinez) [inline client/bindings/java/iota-client-java/CHANGELOG.md:26]:
```suggestion

- Allow integers values in QueryParams;
```

File: client/bindings/java/iota-client-java/CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Changed
+- Improve performance of tests by avoiding unnecessary indexer requests;
```
Comment (thibault-martinez) [inline client/bindings/java/iota-client-java/CHANGELOG.md:30]:
```suggestion

- Improve performance of tests by avoiding unnecessary indexer requests;
```

File: client/bindings/java/iota-client-java/src/main/java/org/iota/apis/NodeIndexerApi.java [hunk 3]
```diff
@@ -114,6 +114,14 @@ public QueryParams withParam(String name, String value) {
             return this;
         }
 
+        public QueryParams withParam(String name, int value) {
+            JsonObject o = new JsonObject();
+            o.addProperty(name, value);
+            queryParams.add(o);
+            return this;
+        }
+
```
Comment (Thoralf-M) [inline client/bindings/java/iota-client-java/src/main/java/org/iota/apis/NodeIndexerApi.java:123]:
```suggestion
```

File: client/bindings/java/iota-client-java/src/test/java/org/iota/ApiTest.java [hunk 4]
```diff
@@ -25,32 +29,44 @@ protected void setUp() {
...
-            } catch (ClientException e) { ; }
+    protected TransactionId setUpTransactionId(String address) throws ClientException {
+        OutputMetadata metadata = client.getOutputMetadata(setupOutputId(address));
+        TransactionId ret = new TransactionId(metadata.toJson().get("transactionId").getAsString());
+        client.getIncludedBlock(ret);
+        return ret;
+    }
+
+    protected OutputId setupOutputId(String address) throws ClientException {
+        OutputId[] outputIds = client.getBasicOutputIds(new NodeIndexerApi.QueryParams().withParam("address", address));
```
Comment (Thoralf-M) [inline client/bindings/java/iota-client-java/src/test/java/org/iota/ApiTest.java]: This is also called in `requestFundsFromFaucet` again, so you can just skip it and just call `requestFundsFromFaucet`?

File: client/bindings/java/iota-client-java/CHANGELOG.md [hunk 5]
```diff
@@ -19,6 +19,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+## 1.0.0-rc.2 - 2022-10-17
```
Comment (thibault-martinez) [inline client/bindings/java/iota-client-java/CHANGELOG.md:22]: Just checking, is there no other file you also need to change this ?

File: client/bindings/java/iota-client-java/src/test/java/org/iota/ApiTest.java (hunk 4 above)
Comment (samuel-rufi) [reply client/bindings/java/iota-client-java/src/test/java/org/iota/ApiTest.java]: Good catch

File: client/bindings/java/iota-client-java/CHANGELOG.md (hunk 5 above)
Comment (samuel-rufi) [reply client/bindings/java/iota-client-java/CHANGELOG.md:22]: Hm not sure I understand, what do you mean?

Comment (thibault-martinez) [reply client/bindings/java/iota-client-java/CHANGELOG.md:22]: Don't you need to bump the version somewhere if you want to release it ?

Comment (samuel-rufi) [reply client/bindings/java/iota-client-java/CHANGELOG.md:22]: Oh, right. Forgot about the build.gradle

File: client/bindings/java/iota-client-java/CHANGELOG.md (hunk 1 above)
Comment (thibault-martinez) [inline client/bindings/java/iota-client-java/CHANGELOG.md:26]:
```suggestion

- Allow integers values in QueryParams;
```

File: client/bindings/java/iota-client-java/CHANGELOG.md (hunk 2 above)
Comment (thibault-martinez) [inline client/bindings/java/iota-client-java/CHANGELOG.md:30]:
```suggestion

- Improve performance of tests by avoiding unnecessary indexer requests;
```

**Review by thibault-martinez — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**